### PR TITLE
[New Rule] File made Immutable by Chattr

### PR DESCRIPTION
**Pull request by DefSecSentinel**
## Issues

https://github.com/elastic/detection-rules/issues/2090

## Summary
Detects a file being made immutable using the chattr binary. Making a file immutable means it cannot be deleted or renamed, no link can be created to this file, most of the file's metadata can not be modified, and the file can not be opened in write mode. Threat actors will commonly utilize this to prevent tampering or modification of their malicious files or any system files they have modified for purposes of persistence (e.g .ssh, /etc/passwd, etc.).

### CODE REVIEW

**Comment by Mikaayenson** [comment]
@DefSecSentinel Can you drop an event here or in the issue?

File: rules/linux/defense_evasion_chattr_immutable_file.toml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+query = '''
+process where event.type == "start" and user.name == "root" and process.executable : "/usr/bin/chattr" and process.args : ("-*i*", "+*i*") and not process.parent.executable: "/lib/systemd/systemd"
+'''
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1222"
+name = "File and Directory Permissions Modifications"
```
Comment (DefSecSentinel) [inline rules/linux/defense_evasion_chattr_immutable_file.toml:37]:
```suggestion
name = "File and Directory Permissions Modification"
```

**Review by terrancedejesus — APPROVED** [review]

**Comment by DefSecSentinel** [comment]
@Mikaayenson 

```json
{
  "_index": "restored-endgame-4.54.2-000036",
  "_id": "pvZ2ZoABzxu7YRCYM8pR",
  "_score": 1,
  "_source": {
    "agent": {
      "name": "Endgame",
      "id": "a7f55140-0453-4eb3-b5d2-626b0aeff168",
      "type": "Endgame",
      "version": "3.61.4"
    },
    "process": {
      "args": [
        "chattr",
        "+i",
        "/var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2"
      ],
      "parent": {
        "args": [
          "sudo",
          "chattr",
          "+i",
          "/var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2"
        ],
        "name": "sudo",
        "pid": 246783,
        "args_count": 4,
        "title": "sudo",
        "executable": "/usr/bin/sudo",
        "command_line": "sudo chattr +i /var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2"
      },
      "code_signature": {
        "exists": false
      },
      "name": "chattr",
      "pid": 246784,
      "args_count": 3,
      "thread": {
        "id": 246784
      },
      "title": "chattr",
      "command_line": "chattr +i /var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2",
      "executable": "/usr/bin/chattr",
      "hash": {
        "sha1": "3b81b93b53ce702c51d85c32ce67190a743a05d5",
        "sha256": "b25e6c167cee869ceda05fe121a5e2a0bf7b79ee4c9d9a62d7034876c9b40664",
        "md5": "09d6c5cd6431a3075e4ff4c1770cd822"
      },
      "ppid": 246783
    },
    "message": "Process exec'd a new image.",
    "observer": {
      "hostname": "x1-home",
      "product": "SMP",
      "os": {
        "name": "Linux",
        "family": "linux",
        "type": "linux",
        "version": "35",
        "platform": "linux"
      },
      "vendor": "Endgame",
      "ip": "192.168.1.154",
      "version": "3.26.0"
    },
    "@timestamp": "2022-04-26T15:14:38.245Z",
    "ecs": {
      "version": "1.9.0"
    },
    "related": {
      "hosts": [
        "x1-home"
      ],
      "ip": [
        "192.168.1.154"
      ],
      "user": [
        "root"
      ],
      "hash": [
        "09d6c5cd6431a3075e4ff4c1770cd822",
        "3b81b93b53ce702c51d85c32ce67190a743a05d5",
        "b25e6c167cee869ceda05fe121a5e2a0bf7b79ee4c9d9a62d7034876c9b40664"
      ]
    },
    "endgame": {
      "sha256": "b25e6c167cee869ceda05fe121a5e2a0bf7b79ee4c9d9a62d7034876c9b40664",
      "unique_ppid": 51123003,
      "pid": 246784,
      "unique_pid": 51123005,
      "opcode": 5,
      "tid": 246784,
      "event_subtype_full": "exec_event",
      "real_group_name": "root",
      "effective_group_name": "root",
      "process_name": "chattr",
      "unix_timestamp": 1650986078245,
      "parent_process_path": "/usr/bin/sudo",
      "timestamp": 132954596782457940,
      "real_gid": 0,
      "parent_command_line": "sudo chattr +i /var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2",
      "effective_user_name": "root",
      "real_user_name": "root",
      "parent_process_name": "sudo",
      "serial_event_id": 51123005,
      "session_id": 227507,
      "event_type_full": "process_event",
      "effective_uid": 0,
      "real_uid": 0,
      "ppid": 246783,
      "sha1": "3b81b93b53ce702c51d85c32ce67190a743a05d5",
      "event_message": "Process exec'd a new image.",
      "process_path": "/usr/bin/chattr",
      "effective_gid": 0,
      "command_line": "chattr +i /var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2",
      "md5": "09d6c5cd6431a3075e4ff4c1770cd822"
    },
    "organization": {
      "name": "Elastic Endgame",
      "id": "8b2b2c86-7b6a-46d9-9441-efe6e4164e3c"
    },
    "host": {
      "hostname": "x1-home",
      "os": {
        "name": "Linux",
        "family": "linux",
        "type": "linux",
        "version": "35",
        "platform": "linux"
      },
      "ip": "192.168.1.154",
      "name": "x1-home",
      "id": "a7f55140-0453-4eb3-b5d2-626b0aeff168",
      "user": {
        "effective": {
          "full_name": "root",
          "name": "root",
          "id": "0",
          "group": {
            "name": "root",
            "id": "0"
          }
        },
        "full_name": "root",
        "name": "root",
        "id": "0",
        "group": {
          "name": "root",
          "id": "0"
        }
      }
    },
    "event": {
      "sequence": 51123005,
      "ingested": "2022-04-26T15:21:03.073424935Z",
      "kind": "event",
      "module": "endgame",
      "action": "exec_event",
      "category": "process",
      "type": "start",
      "dataset": "process"
    },
    "user": {
      "effective": {
        "full_name": "root",
        "name": "root",
        "id": "0",
        "group": {
          "name": "root",
          "id": "0"
        }
      },
      "full_name": "root",
      "name": "root",
      "id": "0",
      "group": {
        "name": "root",
        "id": "0"
      }
    }
  },
  "fields": {
    "endgame.event_type_full": [
      "process_event"
    ],
    "process.hash.md5": [
      "09d6c5cd6431a3075e4ff4c1770cd822"
    ],
    "endgame.sha1": [
      "3b81b93b53ce702c51d85c32ce67190a743a05d5"
    ],
    "event.category": [
      "process"
    ],
    "host.os.name.text": [
      "Linux"
    ],
    "endgame.unique_ppid": [
      51123003
    ],
    "process.parent.command_line": [
      "sudo chattr +i /var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2"
    ],
    "process.parent.name": [
      "sudo"
    ],
    "process.hash.sha256": [
      "b25e6c167cee869ceda05fe121a5e2a0bf7b79ee4c9d9a62d7034876c9b40664"
    ],
    "host.user.effective.full_name": [
      "root"
    ],
    "process.parent.pid": [
      246783
    ],
    "host.hostname": [
      "x1-home"
    ],
    "process.code_signature.exists": [
      false
    ],
    "endgame.real_group_name": [
      "root"
    ],
    "observer.vendor": [
      "Endgame"
    ],
    "host.user.full_name.text": [
      "root"
    ],
    "host.os.version": [
      "35"
    ],
    "user.full_name.text": [
      "root"
    ],
    "user.group.name": [
      "root"
    ],
    "process.parent.title": [
      "sudo"
    ],
    "endgame.timestamp": [
      132954596782457940
    ],
    "host.user.name": [
      "root"
    ],
    "endgame.process_path": [
      "/usr/bin/chattr"
    ],
    "agent.name": [
      "Endgame"
    ],
    "endgame.ppid": [
      246783
    ],
    "observer.os.type": [
      "linux"
    ],
    "host.os.type": [
      "linux"
    ],
    "user.id": [
      "0"
    ],
    "observer.os.platform": [
      "linux"
    ],
    "process.ppid": [
      246783
    ],
    "process.parent.title.text": [
      "sudo"
    ],
    "endgame.opcode": [
      5
    ],
    "host.user.name.text": [
      "root"
    ],
    "related.user": [
      "root"
    ],
    "agent.id": [
      "a7f55140-0453-4eb3-b5d2-626b0aeff168"
    ],
    "observer.version": [
      "3.26.0"
    ],
    "host.user.group.id": [
      "0"
    ],
    "process.command_line.text": [
      "chattr +i /var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2"
    ],
    "user.effective.name.text": [
      "root"
    ],
    "user.name": [
      "root"
    ],
    "endgame.effective_uid": [
      0
    ],
    "endgame.real_user_name": [
      "root"
    ],
    "host.ip": [
      "192.168.1.154"
    ],
    "agent.type": [
      "Endgame"
    ],
    "process.executable.text": [
      "/usr/bin/chattr"
    ],
    "endgame.real_uid": [
      0
    ],
    "related.ip": [
      "192.168.1.154"
    ],
    "observer.product": [
      "SMP"
    ],
    "host.id": [
      "a7f55140-0453-4eb3-b5d2-626b0aeff168"
    ],
    "endgame.md5": [
      "09d6c5cd6431a3075e4ff4c1770cd822"
    ],
    "endgame.event_message": [
      "Process exec'd a new image."
    ],
    "host.user.effective.group.id": [
      "0"
    ],
    "user.effective.full_name.text": [
      "root"
    ],
    "host.user.effective.group.name": [
      "root"
    ],
    "endgame.pid": [
      246784
    ],
    "observer.hostname": [
      "x1-home"
    ],
    "endgame.sha256": [
      "b25e6c167cee869ceda05fe121a5e2a0bf7b79ee4c9d9a62d7034876c9b40664"
    ],
    "endgame.tid": [
      246784
    ],
    "host.user.full_name": [
      "root"
    ],
    "event.action": [
      "exec_event"
    ],
    "event.ingested": [
      "2022-04-26T15:21:03.073Z"
    ],
    "@timestamp": [
      "2022-04-26T15:14:38.245Z"
    ],
    "host.os.platform": [
      "linux"
    ],
    "endgame.parent_process_path": [
      "/usr/bin/sudo"
    ],
    "endgame.unique_pid": [
      51123005
    ],
    "endgame.session_id": [
      227507
    ],
    "process.hash.sha1": [
      "3b81b93b53ce702c51d85c32ce67190a743a05d5"
    ],
    "user.name.text": [
      "root"
    ],
    "endgame.effective_user_name": [
      "root"
    ],
    "endgame.serial_event_id": [
      51123005
    ],
    "process.name.text": [
      "chattr"
    ],
    "endgame.event_subtype_full": [
      "exec_event"
    ],
    "related.hash": [
      "09d6c5cd6431a3075e4ff4c1770cd822",
      "3b81b93b53ce702c51d85c32ce67190a743a05d5",
      "b25e6c167cee869ceda05fe121a5e2a0bf7b79ee4c9d9a62d7034876c9b40664"
    ],
    "user.effective.name": [
      "root"
    ],
    "observer.os.version": [
      "35"
    ],
    "process.pid": [
      246784
    ],
    "process.title.text": [
      "chattr"
    ],
    "endgame.parent_process_name": [
      "sudo"
    ],
    "host.os.name": [
      "Linux"
    ],
    "host.name": [
      "x1-home"
    ],
    "event.kind": [
      "event"
    ],
    "process.parent.args_count": [
      4
    ],
    "process.name": [
      "chattr"
    ],
    "endgame.process_name": [
      "chattr"
    ],
    "ecs.version": [
      "1.9.0"
    ],
    "process.parent.executable.text": [
      "/usr/bin/sudo"
    ],
    "host.user.effective.name": [
      "root"
    ],
    "host.user.group.name": [
      "root"
    ],
    "host.user.effective.id": [
      "0"
    ],
    "endgame.parent_command_line": [
      "sudo chattr +i /var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2"
    ],
    "user.full_name": [
      "root"
    ],
    "organization.id": [
      "8b2b2c86-7b6a-46d9-9441-efe6e4164e3c"
    ],
    "agent.version": [
      "3.61.4"
    ],
    "user.group.id": [
      "0"
    ],
    "related.hosts": [
      "x1-home"
    ],
    "host.os.family": [
      "linux"
    ],
    "process.thread.id": [
      246784
    ],
    "process.title": [
      "chattr"
    ],
    "observer.ip": [
      "192.168.1.154"
    ],
    "user.effective.group.id": [
      "0"
    ],
    "endgame.real_gid": [
      0
    ],
    "user.effective.id": [
      "0"
    ],
    "process.parent.name.text": [
      "sudo"
    ],
    "event.sequence": [
      51123005
    ],
    "observer.os.name.text": [
      "Linux"
    ],
    "event.module": [
      "endgame"
    ],
    "endgame.effective_gid": [
      0
    ],
    "user.effective.group.name": [
      "root"
    ],
    "host.user.id": [
      "0"
    ],
    "process.executable": [
      "/usr/bin/chattr"
    ],
    "endgame.command_line": [
      "chattr +i /var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2"
    ],
    "process.parent.executable": [
      "/usr/bin/sudo"
    ],
    "observer.os.family": [
      "linux"
    ],
    "process.parent.command_line.text": [
      "sudo chattr +i /var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2"
    ],
    "process.args_count": [
      3
    ],
    "user.effective.full_name": [
      "root"
    ],
    "process.args": [
      "chattr",
      "+i",
      "/var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2"
    ],
    "message": [
      "Process exec'd a new image."
    ],
    "organization.name": [
      "Elastic Endgame"
    ],
    "organization.name.text": [
      "Elastic Endgame"
    ],
    "endgame.effective_group_name": [
      "root"
    ],
    "process.parent.args": [
      "sudo",
      "chattr",
      "+i",
      "/var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2"
    ],
    "observer.os.name": [
      "Linux"
    ],
    "event.type": [
      "start"
    ],
    "process.command_line": [
      "chattr +i /var/cache/optimyze/vm-images/optimyze-images/debian-bullseye-ec2-amd64-20211004-785-optimyze-7de4f478-1633441241.qcow2"
    ],
    "endgame.unix_timestamp": [
      1650986078245
    ],
    "event.dataset": [
      "process"
    ]
  }
}
```

**Review by Mikaayenson — APPROVED** [review]
LGTM

**Pull request merged**